appservice/src/intTest/groovy/nu/yona/server/RemoveUserTest.groovy

```diff
@@ -1,159 +1,131 @@
-		given:
-
-		when:
-		def response = appService.getBuddies(bobDunn.url, bobDunn.password);
-
-		then:
-		response.status == 200
-	}
-
-	def 'Bob checks his anonymous messages and will find a remove buddy message'(){
+class RemoveUserTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Delete account'()
+	{
 		given:
+			def richard = addRichard()
 
 		when:
-		def response = appService.getAnonymousMessages(bobDunn.url, bobDunn.password)
-		if (response.responseData._embedded && response.responseData._embedded.buddyDisconnectMessages) {
-			bobDunnBuddyRemoveMessageProcessURL = response.responseData._embedded.buddyDisconnectMessages[0]._links.process.href
-		}
+			def response = appService.deleteUser(richard, "Goodbye friends! I deinstalled the Internet")
 
 		then:
-		response.status == 200
-		response.responseData._embedded.buddyDisconnectMessages[0].reason == "USER_ACCOUNT_DELETED"
-		response.responseData._embedded.buddyDisconnectMessages[0].message == "Goodbye friends! I deinstalled the Internet"
-		response.responseData._embedded.buddyDisconnectMessages[0].nickname == richardQuin.nickname
-		response.responseData._embedded.buddyDisconnectMessages[0]._links.self.href.startsWith(bobDunn.url + appService.ANONYMOUS_MESSAGES_PATH_FRAGMENT)
-		bobDunnBuddyRemoveMessageProcessURL.startsWith(response.responseData._embedded.buddyDisconnectMessages[0]._links.self.href)
+			response.status == 200
 	}
 
-	def 'Bob processes the remove buddy message'(){
+	def 'Delete and recreate account'()
+	{
 		given:
+			def richard = addRichard()
+			appService.deleteUser(richard, "Goodbye friends! I deinstalled the Internet")
 
 		when:
-		def response = appService.postMessageActionWithPassword(bobDunnBuddyRemoveMessageProcessURL, """{
-					"properties":{
-					}
-				}""", bobDunn.password)
+			def newRichard = appService.addUser(appService.&assertUserCreationResponseDetails, richard.password, richard.firstName, richard.lastName,
+				richard.nickname, richard.mobileNumber, [ "Nokia 6310" ], [])
 
 		then:
-		response.status == 200
-		response.responseData.properties.status == "done"
+			newRichard
 	}
 
-	def 'Bob checks his buddy list and will not find Richard there anymore'(){
+	def 'Remove Richard and verify that Bob receives a remove buddy message'()
+	{
 		given:
+			def richardAndBob = addRichardAndBobAsBuddies()
+			def richard = richardAndBob.richard
+			def bob = richardAndBob.bob
+			analysisService.postToAnalysisEngine(bob, ["Gambling"], "http://www.poker.com")
+			analysisService.postToAnalysisEngine(richard, "news/media", "http://www.refdag.nl")
 
 		when:
-		def response = appService.getBuddies(bobDunn.url, bobDunn.password);
+			def message = "Goodbye friends! I deinstalled the Internet"
+			appService.deleteUser(richard, message)
 
 		then:
-		response.status == 200
-		response.responseData._embedded == null
+			// TODO: when are the buddies actually removed? Not on "process" of the message?
```

Comment:
The buddy is removed on process of the message indeed. That is covered by the next tests.